File: frontend/src/types/category.ts
import type { CategoryStatus } from './common'

export interface Category {
  id: number
  parentId: number
  name: string
  icon?: string
  description?: string
  sortOrder: number
  status: CategoryStatus
  postCount: number
  createdAt: string
  updatedAt: string
}

export interface CategoryTree extends Category {
  children?: CategoryTree[]
}

export interface CategoryCreateRequest {
  parentId?: number
  name: string
  icon?: string
  description?: string
  sortOrder?: number
}

export interface CategoryUpdateRequest {
  name?: string
  icon?: string
  description?: string
  sortOrder?: number
}

File: backend/src/main/java/com/techcommunity/entity/Category.java
package com.techcommunity.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_category")
public class Category {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long parentId;

    private String name;

    private String icon;

    private String description;

    private Integer sortOrder;

    private Integer status;

    private Integer postCount;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
